src/classes/HiFiAudioAPIData.ts

```diff
@@ -33,6 +33,30 @@ export class Point3D {
     }
 }
 
+/**
+ * Instantiations of this class define an orientation in 3D space in Quaternion format. A user's orientation in 3D space
+ * affects the way the mixed spatial audio is heard by the user. Additionally, orientation affects the way
+ * a user's audio input propagates through a space: speakers facing directly towards a listener will sound louder than
+ * speakers facing away from a listener.
+ */
+export class OrientationQuat3D {
+    w: number;
+    x: number;
+    y: number;
+    z: number;
+
+    /**
+     * Construct a new `OrientationQuat3D` object. All parameters are required.
+     */
+    constructor({ w = 1, x = 0, y = 0, z = 0 }: { w?: number, x?: number, y?: number, z?: number } = {}) {
```

Comment:
this isn't the same as what's in my branch, and i think this is going to cause problems.

if the last sent quat data component "x" is the same as the current quat data component "x", we won't send the "x" component of the quat up to the server.

check out the diff function on line 248 below. maybe you can fix it there and keep this as the constructor (i like it better this way, if we can make it work)

Reply:
I addressed the diff feature dierectly in the mixer, and sowe don;t need the geometrical structs to be able to have "null" components.
As discussed we prefer point, quat or even euler to have their componetns fully defined